src/js/components/Layer/LayerContainer.js

```diff
@@ -81,22 +81,36 @@ const LayerContainer = forwardRef(
     }, [sendAnalytics, layerRef, position]);
 
     useEffect(() => {
-      if (position !== 'hidden') {
+      if (position !== 'hidden' && modal) {
         const node = layerRef.current || containerRef.current || ref.current;
         if (node && node.scrollIntoView) node.scrollIntoView();
+
         // Once layer is open we make sure it has focus so that you
         // can start tabbing inside the layer. If the caller put focus
         // on an element already, we honor that. Otherwise, we put
         // the focus within the layer. Look at FocusedContainer.js for
         // more details.
         let element = document.activeElement;
+        let isInLayer = false;
         while (element) {
           if (element === containerRef.current) {
-            // already have focus inside the container
+            isInLayer = true;
             break;
           }
           element = element.parentElement;
         }
+        if (!isInLayer) {
+          // Try to focus first focusable child inside the layer
+          const focusable = containerRef.current?.querySelector(
+            'button, [href], input, select, textarea, ' +
```

Comment:
We generally have tried to avoid doing this kind of query selector where possible.

Is there any ability to ensure the focus gets placed on the div with tabIndex 0 that precedes the layer maybe? 

<img width="1434" alt="Screenshot 2025-07-07 at 9 00 23 AM" src="https://github.com/user-attachments/assets/7d9d1c00-23f6-4fe0-b39e-65a2041a5438" />

Reply:
This may or may not be the right approach, but just an idea